src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Formatters/StandardClassLibraryFormatter.cs

```diff
@@ -35,6 +40,581 @@ public byte[] Deserialize(ref MessagePackReader reader, MessagePackSerializerOpt
+                        0, 1, 2, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128, 128, 128,
+                        128, 0, 1, 2, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128, 128,
+                        0, 128, 1, 2, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128, 128,
+                        128, 0, 128, 1, 2, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        0, 1, 128, 2, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128, 128,
+                        128, 0, 1, 128, 2, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        0, 128, 1, 128, 2, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        128, 0, 128, 1, 128, 2, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        0, 1, 2, 128, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128, 128,
+                        128, 0, 1, 2, 128, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        0, 128, 1, 2, 128, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        128, 0, 128, 1, 2, 128, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        0, 1, 128, 2, 128, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        128, 0, 1, 128, 2, 128, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        0, 128, 1, 128, 2, 128, 3, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        128, 0, 128, 1, 128, 2, 128, 3, 4, 128, 5, 128, 6, 128, 7, 128,
+                        0, 1, 2, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128, 128,
+                        128, 0, 1, 2, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        0, 128, 1, 2, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        128, 0, 128, 1, 2, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        0, 1, 128, 2, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        128, 0, 1, 128, 2, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        0, 128, 1, 128, 2, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        128, 0, 128, 1, 128, 2, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128,
+                        0, 1, 2, 128, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128, 128,
+                        128, 0, 1, 2, 128, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        0, 128, 1, 2, 128, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        128, 0, 128, 1, 2, 128, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128,
+                        0, 1, 128, 2, 128, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128, 128,
+                        128, 0, 1, 128, 2, 128, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128,
+                        0, 128, 1, 128, 2, 128, 3, 128, 4, 128, 5, 128, 6, 128, 7, 128,
+                        128, 0, 128, 1, 128, 2, 128, 3, 128, 4, 128, 5, 128, 6, 128, 7,
+                    };
+                    fixed (byte* pShuffle = shuffle)
```

Comment:
I'm very impressed by what you wrote here. But I'm also very concerned about maintainability. I have no idea what any of this does and have never seen code like it before. The array above is a total mystery to me as to where it came from, and why "shuffling" is required baffles me.  If we keep this, IMO we would need a *lot* of code comments and links to docs or blogs that explain it.

Why in the world do we need to do all this when we could do a straight up memcpy of `sbyte[]` to the memory returned from `GetSpan`?

Reply:
>Why in the world do we need to do all this when we could do a straight up memcpy of sbyte[] to the memory returned from GetSpan?

It's incorrect.
According to [current MessagePackWriter.Write(sbyte)](https://github.com/neuecc/MessagePack-CSharp/blob/68638214982c83d15ab0b0020445f9126ab017e3/src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackWriter.cs#L227), Some sbyte values ranges from -33 to -128 have to be encoded and must not be just copied.

Reply:
> I have no idea what any of this does and have never seen code like it before. 

It is known among the C/C++ programmers who use SIMD hardware intrinsics.
This is a good time for C#ers to learn and use SIMD programming.

> The array above is a total mystery to me as to where it came from, and why "shuffling" is required baffles me. 

I will explain where this `shuffle` table came from in the T4 file.

Reply:
I look forward to the .tt file update.

> It's incorrect.
According to current MessagePackWriter.Write(sbyte), Some sbyte values ranges from -33 to -128 have to be encoded and must not be just copied.

Huh. Good point. I guess when we write out `byte[]` we skip the attempted compression of each byte because we precede it with a special msgpack 'binary' header, which we don't do for `sbyte[]`, so I guess we're on the hook to properly encode each one. But that's... really unfortunate and I would encourage folks who want good perf to simply cast their `sbyte[]` as a `byte[]` (using pointers) for faster and more compact encoding. But I guess I get what you're going for. Thanks for explaining.

Reply:
> simply cast their sbyte[] as a byte[] (using pointers) for faster and more compact encoding. 
I did the same thing for personal usage...

Reply:
References
- [Official Intel API Reference](https://software.intel.com/sites/landingpage/IntrinsicsGuide/)
- [The most useful website to understand the SIMD intruction behaviours](https://www.officedaytime.com/simd512e/)
- [.NET Hardware Intrinsics API](https://docs.microsoft.com/ja-jp/dotnet/api/system.runtime.intrinsics.x86?view=netcore-3.1)

Reply:
The `sbyte[]` formatter is optimized differently than the other types...

The below table is measured by the `Int8ArrayBenchmarkMessagePackNoSimdVsMessagePackSimd` benchmark.

|              Method |      Mean |    Error |   StdDev |
|-------------------- |----------:|---------:|---------:|
|       SerializeSimd |  19.59 ms | 0.345 ms | 0.472 ms |
|     SerializeNoSimd | 116.50 ms | 1.306 ms | 1.221 ms |
|   SerializeSimdZero |  10.05 ms | 0.152 ms | 0.127 ms |
| SerializeNoSimdZero |  56.92 ms | 0.635 ms | 0.594 ms |
|    SerializeSimdM32 |  10.11 ms | 0.092 ms | 0.086 ms |
|  SerializeNoSimdM32 |  59.48 ms | 1.171 ms | 2.112 ms |
|    SerializeSimdM33 |  28.11 ms | 0.375 ms | 0.333 ms |
|  SerializeNoSimdM33 |  91.95 ms | 0.790 ms | 0.701 ms |

Reply:
> But that's... really unfortunate and I would encourage folks who want good perf to simply cast their sbyte[] as a byte[] (using pointers) for faster and more compact encoding.

Actually I did same thing.
I change the formatter to cast `sbyte[]` as `byte[]` and encode it.
This is the benchmark result.

|              Method |      Mean |    Error |   StdDev |
|-------------------- |----------:|---------:|---------:|
|       SerializeSimd_ConvertByteArray |  11.05 ms | 0.206 ms | 0.426 ms |
|     SerializeNoSimd | 127.54 ms | 2.472 ms | 2.064 ms |
|   SerializeSimdZero_ConvertByteArray |  11.32 ms | 0.222 ms | 0.247 ms |
| SerializeNoSimdZero |  64.56 ms | 1.243 ms | 2.830 ms |
|    SerializeSimdM32_ConvertByteArray |  10.89 ms | 0.183 ms | 0.290 ms |
|  SerializeNoSimdM32 |  63.10 ms | 1.178 ms | 1.157 ms |
|    SerializeSimdM33_ConvertByteArray |  10.77 ms | 0.150 ms | 0.133 ms |
|  SerializeNoSimdM33 |  97.38 ms | 0.998 ms | 0.885 ms |

My PR is not that bad compared to that which converts to byte[].